3-UsingAPIs/Week2/assignment/ex3-rollAnAce.js

```diff
@@ -12,21 +12,23 @@ Full description at: https://github.com/HackYourFuture/Assignments/blob/main/3-U
 // ! Do not change or remove the next two lines
 const rollDie = require('../../helpers/pokerDiceRoller');
 
-function rollDieUntil(wantedValue) {
-  // TODO: rewrite this function using async/await
-  return rollDie().then((value) => {
-    if (value !== wantedValue) {
-      return rollDieUntil(wantedValue);
+async function rollDieUntil(wantedValue) {
+  while (true) {
+    const value = await rollDie();
+    if (value === wantedValue) {
+      return value;
     }
```

Comment:
It would be better to let the `while` loop itself handle its exit condition.

```suggestion
  let value = await rollDie();
  while (value !== wantedValue) {
    value = await rollDie();
   }
   return value;
```